### ISSUE TITLE
[Bug] Another application is using port 8080

### ISSUE THREAD

**Issue by Radiatorgaming420** [SteamClientHomebrew/Millennium#26]
error getting steams websocketdebuggerurl: unknown exception
![image](https://github.com/ShadowMonster99/millennium-steam-patcher/assets/89734713/88321a84-70dc-47fd-a58a-4e70132e33c8)

**Comment by ShadowMonster99** [comment]
The reason this is happening is because another application is open and using port 8080, which is needed by Steam.

Run the following command in CMD to pin point the conflicting software, and either close it, or uninstall it
```ps
for /f "tokens=5" %a in ('netstat -aon ^| findstr 8080') do wmic process where processId=%a get ExecutablePath
```

After this, make sure you are on Steam's stable version as it currently does not work on beta.
Cheers!

**Comment by ShabbyDragon633** [comment]
Im having a issue.
I've done the command and get told its steamwebhelper. What should i do?

**Comment by ShadowMonster99** [comment]
> Im having a issue.
> 
> I've done the command and get told its steamwebhelper. What should i do?

With steam open when you go to http://localhost:8080 does it load a webpage with some steam windows?

**Comment by ShabbyDragon633** [comment]
Yes

**Comment by mogg21** [comment]
disable gpu accelerated rendering in webviews in settings -> interface
fixed it for me when steamwebhelper was running on port 8080

**Comment by ShabbyDragon633** [comment]
> disable gpu accelerated rendering in webviews in settings

Thx it's working now

**Comment by QWERTYkeyboards124** [comment]
> > Im having a issue.
> > I've done the command and get told its steamwebhelper. What should i do?
> 
> With steam open when you go to http://localhost:8080 does it load a webpage with some steam windows?

It opens nothing for me

**Comment by Tim-Odermatt** [comment]
![image](https://github.com/user-attachments/assets/8a38ec8f-1017-4a4b-a179-382227e9fa50)
I got this error today. 

Disabling GPU acceleration did not fix it.

**Comment by CODEX0827** [comment]
> ![image](https://private-user-images.githubusercontent.com/91256797/397948203-8a38ec8f-1017-4a4b-a179-382227e9fa50.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ4NjYyNDIsIm5iZiI6MTczNDg2NTk0MiwicGF0aCI6Ii85MTI1Njc5Ny8zOTc5NDgyMDMtOGEzOGVjOGYtMTAxNy00YTRiLWExNzktMzgyMjI3ZTlmYTUwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjIyVDExMTIyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgyM2FkMDhlYTA4MzM3NTMwNGE3ZDYwZjc0ZmViMDQ0MzZiZTZjODdiOTAzNWQwNTNlZjk2OWM0ODk2OGQxNGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.NeMAbL-X8J6gUW4S-K1NCv6eTW48xTiplfjuILHcHNk) I got this error today.
> 
> Disabling GPU acceleration did not fix it.

see this video it help my fix this problem
you need to look for the application that is using localport 8080

**Comment by Tim-Odermatt** [comment]
You are my hero:
My App was "AgentService.exe", which was using port 8080.

Which is apart of the "MiniTool® ShadowMaker" software, which got installed in the latest update of "MiniTool® Partition Wizard".

after uninstalling "MiniTool® ShadowMaker" it now runs without the error.

**Comment by CODEX0827** [comment]
> You are my hero: My App was "AgentService.exe", which was using port 8080.
> 
> Which is apart of the "MiniTool® ShadowMaker" software, which got installed in the latest update of "MiniTool® Partition Wizard".
> 
> after uninstalling "MiniTool® ShadowMaker" it now runs without the error.

funny it was the same for me

**Comment by drybiscuitt** [comment]
for me its still not working only thing running is tomcat8 and it is a localport not a RemotePort
 i disabled gpu accelerated rendering but i dont understand what to do with the possible firewall/windows defender thing should i disable defender of something pls help.

**Comment by shdwmtr** [comment]
nothing can be running in local port 8080

**Comment by glektarssza** [comment]
> @shdwmtr The reason this is happening is because another application is open and using port 8080, which is needed by Steam.

That's not entirely true. Port 8080 is not normally used by Steam. I can go spin up a server on my computer right now on port 8080 with Steam running and there's no issues. The reason there's an issue is because Millennium is starting a remote web debugger on port 8080. I assume this is how you're communicating with Steam's CEF instance to do all the modifications, extensions, and such.

See [include/await_pipe.h#L101](https://github.com/SteamClientHomebrew/Millennium/blob/main/include/await_pipe.h#L101).

**Comment by ejalxndr** [comment]
Millennium does *not* start a remote debugger on port 8080, Millennium simply requests one from Steam, and Steam chooses 8080 (which is not changeable without command line arguments), therefor it **is default** given Steams remote debugger is enabled. 

Of course in the context of my message "needed" is assuming you have Millennium installed, and want them **both** Steam & Millennium to work, in which case it **is** "needed"

No correction was needed, I'm aware of how Millennium works as I'm its creator. I simply wrote it like that for brevity.

**Comment by glektarssza** [comment]
@shdwmtr 
> (which is not changeable without command line arguments)

It's the default, yes, but therefore it is changeable so you could change it and make this a non-issue by picking a different, less commonly used port. Am I understanding that correctly?

> No correction was needed

I wasn't attempting to correct you. I'm well aware that as the creator you're intimately aware of how Millennium works. I would hope you are!

> I simply wrote it like that for brevity.

That's what led me to post in the first place. The brevity caused confusion because, in theory, it should be possible to change the port that Steam is opening up for remote debugging. If that's not the case then understandable but wanting to confirm that.

**Comment by ejalxndr** [comment]
@glektarssza 

The port is changeable, but unfortunately not by Millennium itself; so I can’t directly address this issue. 

If you were initially trying to ask whether it’s possible to change the port: yes. You’d need to pass custom command line arguments to Steam to configure what port it should use, though you’d have to always launch steam from a custom shortcut.

Your initial post didn’t come across as if you were trying to find a solution, but instead it seemed like a semantic dispute. (however it seems that wasn’t your intention) 

Due to the structure of Millennium, it loads after Steams remote debugger initializes, therefore it has no way of changing anything.

Cheers! 🙂

**Comment by glektarssza** [comment]
@shdwmtr 

> but instead it seemed like a semantic dispute.

Wasn't my intention at all, sorry for the confusion! It was admittedly quite late in the day when I wrote that original post and I'd had a very stressful day up to that point which is why my tone ended up the way it did. Not that it excuses my behavior. I definitely should have re-read what I wrote before I posted it. Deeply sorry for that!

> Due to the structure of Millennium, it loads after Steams remote debugger initializes, therefore it has no way of changing anything.

Okay, that makes sense. Good to know and thank you for the explanation!

Cheers as well!

**Comment by Thaodan** [comment]
Would be possible to use socket files instead?

**Comment by glektarssza** [comment]
> Would be possible to use socket files instead?

That's actually a good idea, at least on Linux. No idea what the Windows level of support would be for that.

**Comment by ejalxndr** [comment]
@Thaodan 
> Would be possible to use socket files instead?

No, Steam (specifically CEF) is in charge of creating the socket. The underlying chrome remote developer tools protocol depends on the web-socket connection, it's hard-coded as such.

**Comment by maze42d** [comment]
ran into this again today, i suppose you could use zenity to do a little popup for this on linux to make it a little more obvious it's happening before it gets worked around properly?

i'd imagine many people wouldnt think to run steam from the terminal to see what's going on and just give up.

**Comment by ejalxndr** [comment]
@maze42d The whole situation will be addressed in a alpha build that will be released soon. It hooks parts of Steam to force it to use another random open port, instead of 8080.

**Comment by glektarssza** [comment]
@shdwmtr That's awesome news! Huge thanks for putting the work in to make that happen!

**Comment by ejalxndr** [comment]
@maze42d took many long days of brain storming 😅

I forgot to update everyone, but the builds are available right now. keep in mind they are alpha though so they may have bugs. if anyone needs help you can ask here.

**Comment by kotenok2000** [comment]
I has been running it with -devtools-port=8081
